### ISSUE TITLE
`tbot init` should initialize the internal data directory

### ISSUE THREAD

**Issue by timothyb89** [gravitational/teleport#15788]
What would you like Teleport to do?

`tbot init` should initialize (or at least validate) the internal data directory, usually `/var/lib/teleport/bot`.

What problem does this solve?

The internal data directory contains the renewable cert which needs to be as protected as possible, usually owned by a bot (unix) user with 0600 permissions. It should ideally also exist.

If a workaround exists, please include it.

Users can create it manually. rpm and deb packages potentially also create it automatically for regular `teleport` use.

**Comment by strideynet** [comment]
```
What would you like Teleport to do?

Configure the Machine ID data dir in addition to output dir with tbot init.

What problem does this solve?

When running machine id as a non-root user (teleport in our examples) the data dir is not initialized by a Teleport process. That can lead to permission denied on the initial run and requires administrators to setup the data dir. Also if a Teleport service is running in parallel its default permissions on the /var/lib/teleport directory can block access.

If a workaround exists, please include it.

Manually configure the data dir for a teleport user to maintain and set executable for the parent dir.
```

Copied from duplicate ticket https://github.com/gravitational/teleport/issues/22467

**Comment by zmb3** [comment]
@timothyb89 is this still applicable or is it obsolete with the config-less CLI updates?

**Comment by timothyb89** [comment]
Hrm, I think it's technically still relevant, but I don't think it's a great idea to implement. It was probably out of scope for `tbot init` anyway and would just further confuse the UX. And now that `tbot init` is quasi-deprecated I'd rather not implement it as written here.

For context, bots already try to create the internal data directory (`.storage.path` or the `--storage` flag) on startup if it doesn't exist. This will fail if the bot doesn't have permission to create it, which isn't uncommon because the default value is `/var/lib/teleport/bot`. Users will need to create that directory if it doesn't exist, i.e. run `mkdir -p /var/lib/teleport/bot` and `chmod` it as needed, or just change the path to something the bot user can write to.

`tbot init` existed to manage both reader and writer ACLs, and allowed bots, readers, and owners to all be separate users. This was excessively flexible, so we've now (as of #47841) moved to a more opinionated install strategy: bots should own all directories. Instead of running `tbot init`, users can optionally configure readers via the CLI or `tbot.yaml`, and no additional command is needed. Since the bot owns everything, it can manage reader ACLs without having to run anything as root - which `tbot init` needed for more complex situations.

I think if we see end users struggling with this, I'd rather have the error message (when auto creation fails) tell users to run `mkdir ... && chmod ...` than have something try to manage this automagically.
